src/ssl.c

```diff
@@ -17112,6 +17191,25 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
         return 0;
     }
 
+    #if !defined(HAVE_SELFTEST) && (!defined(HAVE_FIPS) || \
+        (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION > 2)))
+    int wolfSSL_SHA512_Transform(WOLFSSL_SHA512_CTX* sha512, 
+                                          const unsigned char* data)
+    {
+       int ret = WOLFSSL_SUCCESS;
+       
+       WOLFSSL_ENTER("SHA512_Transform");
+       
+       ret = wc_Sha512Transform((wc_Sha512*)sha512, data);
```

Comment:
It doesn't look like wc_Sha512Transform() checks sha512 or data against NULL. We should check/sanitize those arguments coming into wolfSSL_SHA512_Transform() first.

Reply:
Exactly. Thank you for pointing it out!